File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/LinuxDialog.cs
namespace AbstractFactoryDesignPattern.Example1;

public class LinuxDialog : Dialog
{
	public override void Render()
	{
		Console.WriteLine("Rendering Linux Dialog");
	}
	public override void Show()
	{
		Console.WriteLine("Showing Linux Dialog");
	}
	public override void Close()
	{
		Console.WriteLine("Closing Linux Dialog");
	}
}

File: java/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/main/java/creational/abstractfactory/example1/models/concretes/LinuxDialog.java
package creational.abstractfactory.example1.models.concretes;

import creational.abstractfactory.example1.models.abstracts.Dialog;

public class LinuxDialog extends Dialog {
    public void render() { System.out.println("Rendering Linux Dialog"); }
    public void show()   { System.out.println("Showing Linux Dialog"); }
    public void close()  { System.out.println("Closing Linux Dialog"); }
}
